### ISSUE TITLE
Row selection loses bright color when focus is lost

### ISSUE THREAD

**Issue by delosbomos** [macgile/DataGridFilter#37]
Hi,
I'm using two DataGridFilters side by side and need to select several rows in both DataGridFilters to perform some actions.
As soon as one of the DGF's losse focus the selection color changes from Highlight color to light gray (almost not visible).

Following solution seems to work for 'normal' datagrids, but not for DataGridFilter, can you help out how to accomplish this?

<DataGrid.Resources>
<SolidColorBrush x:Key="{x:Static SystemColors.InactiveSelectionHighlightBrushKey}"
Color="{Binding Path=Color, Source={x:Static SystemColors.HighlightBrush}}"/>
<Style TargetType="DataGridRow">

<Style.Resources>
<SolidColorBrush x:Key="{x:Static SystemColors.HighlightBrushKey}" Color="Red"/>
</Style.Resources>
</Style>
</DataGrid.Resources>

Kind regards.

**Comment by macgile** [comment]
Hi,

try <control:FilterDataGrid.Resources>

**Comment by delosbomos** [comment]
Hi, I tried your suggestion, but it doesn't work. It throws an exception.
Your demo application does the same, as soon as you start selecting rows in de bottom table, the selection you made in 
the top table is almost invisible (but still there).

I added the following lines to your demo application, but I get the same 'exception'.

----------------------------------
<control:FilterDataGrid.Resources>
<SolidColorBrush x:Key="{x:Static SystemColors.InactiveSelectionHighlightBrushKey}" Color="{Binding Path=Color, Source={x:Static SystemColors.HighlightBrush}}"/>
<Style TargetType="DataGridRow">
<Style.Resources>
<SolidColorBrush x:Key="{x:Static SystemColors.HighlightBrushKey}" Color="Red"/>
</Style.Resources>
</Style>
</control:FilterDataGrid.Resources>
------------------------------------

I think part of the code has to be implemented in de filterdatagrid.xaml, but I leg xaml skills.
Might this be an enhanced for the filterdatagrid ?

Best regards,
Guido van den Boom

**Comment by macgile** [comment]
I don't understand what you want to do, provide me with screenshots or a working app, maybe I can help you.

**Comment by delosbomos** [comment]
I provided 3 screenshots from the demo application.
I hope you understand better what I would like to achieve.

1. I selected some rows from the top table.
2. I selected some rows from the bottom table, but the selection from the top table is not (barely) visible anymore. When selecting more rows in the top table, then the selection from the bottom table is not visible.
3. I assembled a picture what I would like........... the selection from the top table still highlighted when rows from the bottom table are being selected.

Thanx for your help sofar!

![1  selecting rows in top table](https://user-images.githubusercontent.com/86485054/144406860-4840e265-1001-4bb5-b596-340ef1914163.png
![2  selecting rows in bottom table](https://user-images.githubusercontent.com/86485054/144406900-3d72c42a-83c7-4a7a-a099-4874dc26486f.png)
![3  preferred solution (top table selection visible)](https://user-images.githubusercontent.com/86485054/144406936-d856dd79-6c23-4c3e-b4b3-0843ab765529.png)
)

**Comment by macgile** [comment]
It seems to me that this stems from the loss of focus of the DataGrid. 
After you select rows in one data grid and then select other rows in another data grid, the first one loses focus.

**Comment by delosbomos** [comment]
You are right.

With two 'normal' datagrids this can be solved by adding the lines in my first message.
The same lines result in an exception in your filterdatagrid.

Maybe you can incorporate the same?  Or you know where to change your xaml (template) to get the same result?

Best regards

**Comment by macgile** [comment]
Attach a demo app as a zip, I'll see what I can do.

**Comment by delosbomos** [comment]
I just managed to work it out, altough I have no clue how it works.
I just copied and pasted some parts in your App.xaml (see yellow highlighted parts).
The working result is in the 2nd picture.

![image](https://user-images.githubusercontent.com/86485054/144426061-a17eb047-c7ad-40a4-97cc-9ffe34bd8427.png)
![image](https://user-images.githubusercontent.com/86485054/144426391-2c428ca3-4b5d-42d2-b43c-418bc1274e45.png)

Thanx for your time.
